upup/pkg/fi/cloudup/awstasks/network_load_balancer.go

```diff
@@ -0,0 +1,797 @@
+	}
+
+	lbDnsName := fi.StringValue(lb.DNSName)
+	if lbDnsName == "" {
+		return nil, nil
+	}
+	return &lbDnsName, nil
+}
+
+func (e *NetworkLoadBalancer) Run(c *fi.Context) error {
+	// TODO: Make Normalize a standard method
+	e.Normalize()
+
+	return fi.DefaultDeltaRunMethod(e, c)
+}
+
+func (e *NetworkLoadBalancer) Normalize() {
+	// We need to sort our arrays consistently, so we don't get spurious changes
+	sort.Stable(OrderSubnetsById(e.Subnets))
+}
+
+func (s *NetworkLoadBalancer) CheckChanges(a, e, changes *NetworkLoadBalancer) error {
+	if a == nil {
+		if fi.StringValue(e.Name) == "" {
+			return fi.RequiredField("Name")
+		}
+		if len(e.Subnets) == 0 {
+			return fi.RequiredField("Subnets")
+		}
+
+		if e.CrossZoneLoadBalancing != nil {
+			if e.CrossZoneLoadBalancing == nil {
+				return fi.RequiredField("CrossZoneLoadBalancing")
+			}
+		}
+	} else {
+		if len(changes.Subnets) > 0 {
+			return fi.FieldIsImmutable(e.Subnets, a.Subnets, field.NewPath("Subnets"))
+		}
+	}
+	return nil
+}
+
+func (_ *NetworkLoadBalancer) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *NetworkLoadBalancer) error {
+	var loadBalancerName string
+	var loadBalancerArn string
+
+	if a == nil {
+		if e.LoadBalancerName == nil {
+			return fi.RequiredField("LoadBalancerName")
+		}
+		if e.TargetGroup.ARN == nil {
+			return fmt.Errorf("missing required target group ARN for NLB creation %v", e.TargetGroup)
+		}
+
+		loadBalancerName = *e.LoadBalancerName
+
+		request := &elbv2.CreateLoadBalancerInput{}
+		request.Name = e.LoadBalancerName
+		request.Scheme = e.Scheme
+		request.Type = e.Type
+
+		for _, subnet := range e.Subnets {
+			request.Subnets = append(request.Subnets, subnet.ID)
+		}
+
+		{
+			klog.V(2).Infof("Creating NLB with Name:%q", loadBalancerName)
+
+			response, err := t.Cloud.ELBV2().CreateLoadBalancer(request)
+			if err != nil {
+				return fmt.Errorf("error creating NLB: %v", err)
+			}
+
+			if len(response.LoadBalancers) != 1 {
+				return fmt.Errorf("Either too many or too few NLBs were created, wanted to find %q", loadBalancerName)
+			} else {
+				lb := response.LoadBalancers[0]
+				e.DNSName = lb.DNSName
+				e.HostedZoneId = lb.CanonicalHostedZoneId
+				e.VPC = &VPC{ID: lb.VpcId}
+				loadBalancerArn = fi.StringValue(lb.LoadBalancerArn)
+			}
+		}
+
+		{
+			for _, listener := range e.Listeners {
+				createListenerInput := listener.mapToAWS(*e.TargetGroup.ARN, loadBalancerArn)
+
+				klog.V(2).Infof("Creating Listener for NLB")
+				_, err := t.Cloud.ELBV2().CreateListener(createListenerInput)
+				if err != nil {
+					return fmt.Errorf("error creating listener for NLB: %v", err)
+				}
+			}
+		}
+	} else {
+		loadBalancerName = fi.StringValue(a.LoadBalancerName)
+
+		lb, err := findNetworkLoadBalancerByLoadBalancerName(t.Cloud, loadBalancerName)
+		if err != nil {
+			return fmt.Errorf("error getting load balancer by name: %v", err)
+		}
+
+		// if lb == nil {
+		// 	return fmt.Errorf("error querying nlb: %v", err)
+		// }
```

Comment:
This should address the of adding cloudLabels which errors with:
```
W1031 15:54:16.112405   88154 executor.go:131] error running task
"NetworkLoadBalancer/api-k8s-deps-test-bla-com" (9m46s remaining to succeed):
unexpected error fetching tags for resource: error listing tags on :
ValidationError: An ARN must be specified
```
```suggestion
		loadBalancerArn = fi.StringValue(lb.LoadBalancerArn)
```